frontend/public/components/catalog-items.jsx

```diff
@@ -169,26 +209,104 @@ export class CatalogTileViewPage extends React.Component {
+    }
+
+    if (byType.imageStream.active) {
+      filteredItems = filteredItems.concat(_.filter(items, (item) => item.kind === byType.imageStream.value));
+    }
+
+    filteredItems = !_.isEmpty(filteredItems) ? filteredItems : items;
+
+    if (byName.active) {
+      const filterString = byName.value.toLowerCase();
+      return _.filter(filteredItems, item => item.tileName.toLowerCase().includes(filterString));
+    }
+
+    return filteredItems;
+  }
+
+  clearFilters() {
+    const filters = { ...this.state.filters };
+    filters.byName.active = filters.byType.clusterServiceClass.active = filters.byType.imageStream.active = false;
+    filters.byName.value = '';
+    this.setState({ filters });
+  }
+
+  onFilterChange(filterType, id, value) {
+    const filters = { ...this.state.filters };
+    if (filterType === 'byName') {
+      filters[filterType].active = _.size(value) ? true : false;
+      filters[filterType].value = value;
+    } else {
+      filters[filterType][id].active = value;
+    }
+    this.setState({ filters });
+  }
+
   render() {
-    const { activeTabs, showAllItemsForCategory, currentCategories, numItems } = this.state;
+    const { activeTabs, showAllItemsForCategory, currentCategories, numItems, filters } = this.state;
+    const { clusterServiceClass, imageStream } = filters.byType;
     const activeCategory = showAllItemsForCategory ? _.find(currentCategories, { id: showAllItemsForCategory }) : null;
     const heading = activeCategory ? activeCategory.label : this.getCategoryLabel(_.first(activeTabs));
 
     return (
       <div className="co-catalog-page">
         <div className="co-catalog-page__tabs">
           { this.renderCategoryTabs() }
+          <FilterSidePanel id="filter-panel">
+            <FilterSidePanel.Category>
+              <FormControl type="text" placeholder="Filter by name..." bsClass="form-control"
+                value={filters.byName.value}
+                onChange={e => this.onFilterChange('byName', null, e.target.value)}
+              />
+            </FilterSidePanel.Category>
+            <FilterSidePanel.Category title="Type">
+              <FilterSidePanel.CategoryItem
+                count={clusterServiceClass.count}
+                checked={clusterServiceClass.active}
+                onChange={e => this.onFilterChange('byType', 'clusterServiceClass', e.target.checked)}
+              >
+                {clusterServiceClass.label}
+              </FilterSidePanel.CategoryItem>
+              <FilterSidePanel.CategoryItem
+                count={imageStream.count}
+                checked={imageStream.active}
+                onChange={e => this.onFilterChange('byType', 'imageStream', e.target.checked)}
+              >
+                {imageStream.label}
+              </FilterSidePanel.CategoryItem>
+            </FilterSidePanel.Category>
+          </FilterSidePanel>
```

Comment:
Wondering if catalog-items.jsx is getting too big?  Should I move filter stuff to `catalog-filter-sidepanel.tsx` separate file?